src/Disks/IDiskRemote.h

```diff
@@ -0,0 +1,250 @@
+#pragma once
+#include <Common/config.h>
+
+#if USE_AWS_S3
+
+#include <atomic>
+#include "Disks/DiskFactory.h"
+#include "Disks/Executor.h"
+#include <Poco/DirectoryIterator.h>
+#include <utility>
+#include <aws/s3/model/CopyObjectRequest.h>
+#include <aws/s3/model/DeleteObjectsRequest.h>
+#include <aws/s3/model/GetObjectRequest.h>
+#include <aws/s3/model/ListObjectsV2Request.h>
+#include <aws/s3/model/HeadObjectRequest.h>
+#include <Common/MultiVersion.h>
+
+
+namespace DB
+{
+
+/// Helper class to collect keys into chunks of maximum size (to prepare batch requests to AWS API).
+/// Used for both S3 and HDFS.
+class RemoteFSPathKeeper : public std::list<Aws::Vector<Aws::S3::Model::ObjectIdentifier>>
```

Comment:
We shouldn't use S3-related things here. Should be `std::list<std::vector<std::string>>`

Reply:
Fixed, but in a little bit different way, because in `Aws::S3::Model::Delete delkeys; delkeys.SetObjects(chunk);` it is not possible to initialize `delkeys` with `std::vector<std::string>>` or `std::vector<const chat *>`.